### PR TITLE
refactor!: Remove `Measurement` from Core

### PR DESCRIPTION
**Pull request by andiwand**
This was relatively long on our wishlist and we hope to be able to make it for v36. The idea here is that `Acts` does not really need a measurement EDM since we abstract this away with `SourceLink` for uncalibrated measurements and `TrackState` for calibrated measurements. A calibrated measurement is transformed using the uncalibrated measurement and the track parameter at a given surface therefore it cannot be reused for other tracks.

### CODE REVIEW

**Comment by andiwand** [comment]
cc @paulgessinger

File: Examples/Framework/include/ActsExamples/EventData/Measurement.hpp [hunk 1]
```diff
@@ -1,21 +1,254 @@
...
+
+  std::ostream& operator<<(std::ostream& os) const {
+    Acts::detail::printMeasurement(os, static_cast<indices_t>(kSize),
+                                   m_subspace.indices().data(), m_params.data(),
+                                   m_cov.data());
+    return os;
+  }
+
+ private:
+  Acts::SourceLink m_source;
```
Comment (paulgessinger) [inline Examples/Framework/include/ActsExamples/EventData/Measurement.hpp:163]: Should this even have a source link itself? Are we accessing this at all?

**Review by paulgessinger — APPROVED** [review]
Looks good to me,

**Review by paulgessinger — DISMISSED** [review]
/cc @fredevb re: removal of `Acts::Measurement` from Core.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [749489f2d806427a376fa163e3f434f262791c2a]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40180710) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7588258)!

File: Examples/Framework/include/ActsExamples/EventData/Measurement.hpp (hunk 1 above)
Comment (andiwand) [reply Examples/Framework/include/ActsExamples/EventData/Measurement.hpp:163]: I think we can remove this

Comment (andiwand) [reply Examples/Framework/include/ActsExamples/EventData/Measurement.hpp:163]: This seems to be used in some writer. Maybe we can push this to a second iteration if necessary?

**Comment by andiwand** [comment]
Prepared a patch for this on the canary - should be ready now

**Comment by andiwand** [comment]
This seems to work with Athena now @paulgessinger. `test_ActsWorkflow` seems to be a random failure

**Review by paulgessinger — APPROVED** [review]
LGTM

**Pull request merged**